tests/lax_autodiff_test.py

```diff
@@ -31,15 +31,28 @@
 from jax import test_util as jtu
 from jax.test_util import check_grads
 
-from tests.lax_test import (CombosWithReplacement, compatible_shapes,
-                            complex_dtypes, float_dtypes, inexact_dtypes,
-                            num_float_bits)
-
 from jax.config import config
 config.parse_flags_with_absl()
 FLAGS = config.FLAGS
 
 
+def supported_dtypes(dtypes):
+  return [t for t in dtypes if t in jtu.supported_dtypes()]
+
+float_dtypes = supported_dtypes([dtypes.bfloat16, onp.float16, onp.float32,
+                                 onp.float64])
+complex_elem_dtypes = supported_dtypes([onp.float32, onp.float64])
+complex_dtypes = supported_dtypes([onp.complex64, onp.complex128])
+inexact_dtypes = float_dtypes + complex_dtypes
+int_dtypes = supported_dtypes([onp.int32, onp.int64])
+uint_dtypes = supported_dtypes([onp.uint32, onp.uint64])
+bool_dtypes = [onp.bool_]
+default_dtypes = float_dtypes + int_dtypes
+all_dtypes = float_dtypes + complex_dtypes + int_dtypes + bool_dtypes
```

Comment:
Maybe put this in test_util.py? I feel like this comes up a lot (even if I only see it in two places here. Maybe there's more to dedup later).

Reply:
I've been working on that in #3561, but it will have to be defined lazily because the device is not known at import time within `test_util.py`. Still thinking about how to best do that.